tools/AzureRM.BootStrapper/AzureRM.Bootstrapper.psm1

```diff
@@ -838,26 +829,33 @@ function Get-AzureRmProfile
     if ($ListAvailable.IsPresent)
     {
       Write-Verbose "Getting all the profiles available for install"
-      Get-ProfilesAvailable $ProfileMap
+      foreach ($profile in ($ProfileMap | get-member -MemberType NoteProperty).Name)
+      {
+        $profileObj = $ProfileMap.$profile
+        $profileObj | Add-Member -MemberType NoteProperty -Name "ProfileName" -Value $profile
```

Comment:
Yes, this is intentional. This will push an array of profile objects to output stream. If we push one profilemap object (with all the profiles), we wouldn't be able to display all of them using format.ps1xml file because of the above mentioned limitation. So we decided to separately push the profile objects in to the stream